myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/yarn/interceptor/CompositeInterceptor.java

```diff
@@ -42,13 +42,8 @@ public void setMyriadInitInterceptor(YarnSchedulerInterceptor myriadInitIntercep
 
     @Override
     public void register(YarnSchedulerInterceptor interceptor) {
-        if (interceptors.containsKey(interceptor.getClass())) {
-            throw new RuntimeException("More than one instance of " +
```

Comment:
I am guessing this code was added for debugging. I expect most business logic objects to be loaded using Guice as singleton. So we don't really need this check. Besides, I hit an issue in unit test when it tries to invoke YARN scheduler's reinitialize method. If this method gets invoked in production use case, then it will cause Myriad to fail because of this check.